website/pages/blog/complete-guide-to-graphql-scalar.mdx

```diff
@@ -0,0 +1,423 @@
+  Int: { input: number; output: number };
+  Float: { input: number; output: number };
+  CustomScalar: { input: string; output: number };
+};
+```
+
+The recommended server config makes it more convenient to handle ID scalar types. This approach is particularly useful if you have objects with numeric IDs. Without using the recommended config, we would have to create custom mappers or manually convert the number to string to satisfy TypeScript typecheck.
+
+
+<Callout type="info">
+  The recommeded setup is suitable for the most common use cases. It does not cover some edge cases that may make writing resolvers awkward. 
+  
+  For example, `Int`'s output is technically `string | number | boolean` but it is typed as `number` by default because most resolvers would almost never return `string` or `boolean`.
+</Callout>
+
+### 2. GraphQL Scalars (`graphql-scalars`)
+
+[GraphQL Scalars](https://the-guild.dev/graphql/scalars) is a library that contains many commonly used custom GraphQL Scalars such as `DateTime`, `BigInt`, etc. 
+
+Even if it is not too hard to create custom scalars, testing and publishing scalars to share between GraphQL servers can quickly distract from building the core business logic. Therefore, it is usually better to use a well-maintained and documented library like `graphql-scalars`.
+
+Here's how we can use `DateTime` scalar from `graphql-scalars`:
+
+1. Install `graphql-scalars`:
+
+```shell
+yarn add graphql-scalars
+```
+
+2. Declare `DateTime` in the schema:
+
+```graphql
+# schema.graphql
+scalar DateTime
+```
+
+3. Import `DateTime` resolver into resolvers map:
+
+```tsx
+import { createYoga, createSchema } from "graphql-yoga";
+import { createServer } from "http";
+import { DateTimeResolver } from "graphql-scalars"; // 1. Import scalar resolver
+
+const yoga = createYoga({
+  schema: createSchema({ 
+    typeDefs: /* GraphQL typeDefs */, 
+    resolvers: {
+      DateTime: DateTimeResolver // 2. Put resolver to resolvers map
+    }
+  }),
+});
+const server = createServer(yoga);
+server.listen(4000, () => {
+  console.info('Server is running on http://localhost:4000/graphql')
+});
+
+```
+
+4. Use GraphQL Code Generator to create types:
+
+Most scalar resolvers from `graphql-scalars` come with recommended server type that can be used in Codegen config. We can import a resolver and use said type very easily:
+
+```tsx
+// codegen.ts
+import type { CodegenConfig } from '@graphql-codegen/cli';
+import { DateTimeResolver } from "graphql-scalars"; // 1. Import scalar resolver
+
+const config: CodegenConfig = {
+  schema: '**/schema.graphql',
+  generates: {
+    'src/schema/types.generated.ts': {
+      plugins: ['typescript', 'typescript-resolvers'],
+      config: {
+        scalars: {
+          ID: {
+            input: 'string',
+            output: 'string | number',
+          },
+          DateTime: DateTimeResolver.extensions.codegenScalarType // 2. Use scalar resolver's recommended server type
```

Comment:
Is this the right way of using the extensions type? 🤔 

There's something in the implementation here but I'm not 100% sure how it works: https://github.com/dotansimha/graphql-code-generator/blob/9e8bd9d1039646217ef0e6007e3a349631bdd1a3/packages/plugins/other/visitor-plugin-common/src/utils.ts#L358-L367